src/gui/layout/qgslayoutscalebarwidget.cpp

```diff
@@ -56,6 +56,26 @@ QgsLayoutScaleBarWidget::QgsLayoutScaleBarWidget( QgsLayoutItemScaleBar *scaleBa
   connect( mMinWidthSpinBox, static_cast < void ( QDoubleSpinBox::* )( double ) > ( &QDoubleSpinBox::valueChanged ), this, &QgsLayoutScaleBarWidget::mMinWidthSpinBox_valueChanged );
   connect( mMaxWidthSpinBox, static_cast < void ( QDoubleSpinBox::* )( double ) > ( &QDoubleSpinBox::valueChanged ), this, &QgsLayoutScaleBarWidget::mMaxWidthSpinBox_valueChanged );
   connect( mNumberFormatPushButton, &QPushButton::clicked, this, &QgsLayoutScaleBarWidget::changeNumberFormat );
+
+  connect( mSegmentsLeftDDBtn, &QgsPropertyOverrideButton::activated, mSegmentsLeftSpinBox, &QSpinBox::setDisabled );
+  registerDataDefinedButton( mSegmentsLeftDDBtn, QgsLayoutObject::ScalebarLeftSegments );
+  connect( mSegmentsRightDDBtn, &QgsPropertyOverrideButton::activated, mNumberOfSegmentsSpinBox, &QSpinBox::setDisabled );
+  registerDataDefinedButton( mSegmentsRightDDBtn, QgsLayoutObject::ScalebarRightSegments );
+  connect( mSegmentSizeDDBtn, &QgsPropertyOverrideButton::activated, mSegmentSizeSpinBox, &QDoubleSpinBox::setDisabled );
+  registerDataDefinedButton( mSegmentSizeDDBtn, QgsLayoutObject::ScalebarSegmentWidth );
+  connect( mMinWidthDDBtn, &QgsPropertyOverrideButton::activated, mMinWidthSpinBox, &QDoubleSpinBox::setDisabled );
+  registerDataDefinedButton( mMinWidthDDBtn, QgsLayoutObject::ScalebarMinWidth );
+  connect( mMaxWidthDDBtn, &QgsPropertyOverrideButton::activated, mMaxWidthSpinBox, &QDoubleSpinBox::setDisabled );
+  registerDataDefinedButton( mMaxWidthDDBtn, QgsLayoutObject::ScalebarMaxWidth );
+
+  /*
+  mSegmentsLeftDDBtn->registerLinkedWidget( mSegmentsLeftSpinBox );
+  mSegmentsRightDDBtn->registerLinkedWidget( mNumberOfSegmentsSpinBox );
```

Comment:
Can you explain what's happening here?

Reply:
I was hoping someone would bring this up. I was poking around in other classes to figure out what I needed to do to add this feature, and I saw this `registerLinkedWidget()` function with very vague function description used elsewhere. I didn't notice any difference with or without these lines. I just commented them out in case they were necessary for something and needed to be put back.

Reply:
Well, it's duplicate logic to your manual connects just above this, and it would be better to use the standard method over the manual connects

Reply:
To use the `registerLinkedWidget()`, do I remove both the `connect()` and `registerDataDefinedButton()` functions, or just one of them?